client/src/test/scala/org/http4s/client/PoolManagerSpec.scala

```diff
@@ -70,6 +72,28 @@ class PoolManagerSpec(name: String) extends Http4sSpec {
       } yield ()).unsafeRunTimed(2.seconds) must_== Some(())
     }
 
+    // this is a regression test for https://github.com/http4s/http4s/issues/2962
+    "fail expired connections and then wake up a non-expired waiting connection on release" in {
+      val timeout = 50.milliseconds
+      (for {
+        pool <- mkPool(maxTotal = 1, maxWaitQueueLimit = 3, requestTimeout = timeout)
+        conn <- pool.borrow(key)
+        waiting1 <- pool.borrow(key).start
+        waiting2 <- pool.borrow(key).start
+        _ <- IO.sleep(timeout + 20.milliseconds)
+        waiting3 <- pool.borrow(key).start
+        _ <- pool.release(conn.connection)
+        result1 <- waiting1.join.void.attempt
+        result2 <- waiting2.join.void.attempt
+        result3 <- waiting3.join.void.attempt
+      } yield (result1, result2, result3)).unsafeRunTimed(10.seconds) must beSome.like {
```

Comment:
This, and the rest of this suite, should do better in Travis CI when we have cats-effect-testing, but this branch is cats-effect-1, so I'm just bumping the timeout higher to try to survive.